ext/curl/lib/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+target_compile_definitions(app PRIVATE HAVE_CONFIG_H CURL_NO_OLDIES)
+
+target_sources(app PRIVATE 
```

Comment:
Why is this placed in app lib ?
Looks very wrong to me.

App lib belongs to the application.

Reply:
We've just moved curl from modem_shell/src to /ext two days ago (earlier decision to keep curl within the app was reverted as part of the PR review) so apparently we should change this.

Reply:
I've changed curl CMake files now significantly. Our intention is to have curl as a normal library like getopt and iperf3 so anything that doesn't look good from that perspective is a mistake/bug.
We are not experts on CMake files and as we just moved Curl outside of modem shell sample, things might be very wrong.
Please take a look at the latest files which should be much better aligned with iperf3 and getopt.

Reply:
Thanks for the explanation, and knowing the background and reason makes it easier to reiview and provide proper feedback 👍